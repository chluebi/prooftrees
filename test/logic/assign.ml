open Base.Trees
open Logic
open Logic.Examples

let () =
  let inputs =
    [
      (tree1, ass0); (tree1, ass1); (tree1, ass2); (tree3, ass3); (tree4, ass3);
    ]
  in
  let module ResultModule = TreeAssigner (LogicExpression) in
  List.iter
    (fun (t, a) ->
      print_endline (ResultModule.print_tree (ResultModule.assign a t)))
    inputs
