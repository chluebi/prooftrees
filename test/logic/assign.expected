Or(A, B)
Or(A, And(C, D))
Or(Or(A, B), And(C, D))
Or(Or(A, A), Or(A, A))
Not(Or(A, A))
